src/process.rs

```diff
@@ -310,46 +198,411 @@ impl Process {
+        });
+        match logger {
+            Ok(logger) => Ok(Self(logger)),
+            Err(err) => {
+                error!("Cannot connect to syslog: {}", err);
+                Err(Failed)
+            }
+        }
+    }
+
+    /// Tries logging.
+    fn log(&mut self, record: &log::Record) -> Result<(), io::Error> {
+        match record.level() {
+            log::Level::Error => self.0.err(record.args()),
+            log::Level::Warn => self.0.warning(record.args()),
+            log::Level::Info => self.0.info(record.args()),
+            log::Level::Debug => self.0.debug(record.args()),
+            log::Level::Trace => self.0.debug(record.args()),
+        }.map_err(|err| {
+            match err.0 {
+                syslog::ErrorKind::Io(err) => err,
+                syslog::ErrorKind::Msg(err) => {
+                    io::Error::new(io::ErrorKind::Other, err)
+                }
+                err => {
+                    io::Error::new(io::ErrorKind::Other, format!("{}", err))
+                }
+            }
+        })
+    }
+
+    /// Flushes the logger.
+    ///
+    /// Ignores any errors.
+    fn flush(&mut self) {
+        let _ = self.0.backend.flush();
+    }
+}
+
+
+//------------ GlobalLogger --------------------------------------------------
+
+/// The global logger.
+///
+/// A value of this type can go into a static. Until a proper logger is
+/// installed, it just writes all log output to stderr.
+struct GlobalLogger {
+    /// The real logger. Can only be set once.
+    inner: OnceCell<Logger>,
+}
+
+/// The static for the log crate.
+static GLOBAL_LOGGER: GlobalLogger = GlobalLogger::new();
+
+impl GlobalLogger {
+    /// Creates a new provisional logger.
+    const fn new() -> Self {
+        GlobalLogger { inner: OnceCell::new() }
+    }
+
+    /// Switches to the proper logger.
+    fn switch(&self, logger: Logger) {
+        if self.inner.set(logger).is_err() {
+            panic!("Tried to switch logger more than once.")
+        }
+    }
+
+    /// Performs a log rotation.
+    fn rotate(&self) -> Result<(), Failed> {
+        match self.inner.get() {
+            Some(logger) => logger.rotate(),
+            None => Ok(()),
+        }
+    }
+}
+
+
+impl log::Log for GlobalLogger {
+    fn enabled(&self, _: &log::Metadata<'_>) -> bool {
+        true
+    }
+
+    fn log(&self, record: &log::Record<'_>) {
+        match self.inner.get() {
+            Some(logger) => logger.log(record),
+            None => {
+                let _ = writeln!(
+                    io::stderr().lock(), "[{}] {}",
+                    record.level(), record.args()
+                );
+            }
+        }
+    }
+
+    fn flush(&self) {
+        if let Some(logger) = self.inner.get() {
+            logger.flush()
+        }
+    }
+}
+
+
 //------------ LogOutput -----------------------------------------------------
 
 #[derive(Debug)]
 pub struct LogOutput {
```

Comment:
So this is only used to optionally serve logging over HTTP, correct?
(I guess what I'm saying is, this part could benefit from some docstrings)